atom/browser/api/atom_api_app.cc

```diff
@@ -1154,9 +1154,13 @@ v8::Local<v8::Promise> App::GetGPUInfo(v8::Isolate* isolate,
                                        const std::string& info_type) {
   auto* const gpu_data_manager = content::GpuDataManagerImpl::GetInstance();
   scoped_refptr<util::Promise> promise = new util::Promise(isolate);
-  if ((info_type != "basic" && info_type != "complete") ||
-      !gpu_data_manager->GpuAccessAllowed(nullptr)) {
-    promise->Reject("Error fetching GPU Info");
+  if (info_type != "basic" && info_type != "complete") {
+    promise->Reject("Invalid info type. Use 'basic' or 'complete'");
+    return promise->GetHandle();
+  }
+  std::string reason;
+  if (!gpu_data_manager->GpuAccessAllowed(&reason)) {
+    promise->Reject("GPU access not allowed. Reason: " + reason);
```

Comment:
> Error: the string "GPU access not allowed. Reason: GPU process crashed too many times with SwiftShader." was thrown, throw an Error :)

Btw, you should reject a promise with an Error, not a string.

Reply:
There is a helper for this on the promise util, it will construct the error object for you 👍 

```cpp
promise->RejectWithErrorMessage("my message")
```